src/app/model/ChessEngine.ts

```diff
@@ -14,8 +16,10 @@ export class ChessEngine implements ChessModel {
     this.squaresWithPiece = board.squaresWithPiece;
   }
 
-  move(squareFrom: Square, squareTo: Square): (PieceWasMoved | PieceWasCaptured)[] {
+  move(squareFrom: Square, squareTo: Square): ChessboardEvent[] {
+    const events = [];
     const chosenPiece = this.board.onPositionPiece(squareFrom);
+    //TODO warunek który sprawdza  czy nie chcemy wystawić własnego króla
```

Comment:
'wystawić własnego króla' - szkoda, że nie kodujemy po Polsku :D 
Jak byśmy nie zrobili tego feature, bo by było:
- Ale tutaj można swojego króla wystawić!
- To nie bug, to feature - szachy ze zdradą :D

Reply:
:D